core/HoistComponent.ts

```diff
@@ -328,12 +328,18 @@ function createModel(spec) {
 }
 
 function lookupModel(spec, props, modelLookup, displayName) {
-    let {model} = props,
+    let {model, modelConfig} = props,
         {selector} = spec;
 
     // 1) props - config
-    if (model && isPlainObject(model) && spec.createFromConfig) {
-        return {model: new selector(model), isLinked: true, fromContext: false};
+    if (spec.createFromConfig) {
+        if (isPlainObject(model)) {   // 1a) legacy, pre-typescript
+            apiDeprecated('model', {msg: "Use 'modelConfig' instead. ", v: 'v55'});
+            return {model: new selector(model), isLinked: true, fromContext: false};
+        }
+        if (isPlainObject(modelConfig)) {  // 1b) new location
```

Comment:
Should we add `modelConfig` to `HoistProps`? Or to `TabContainerProps`? It all seems a bit mysterious from the app developer's perspective